### ISSUE TITLE
Versioning

### ISSUE THREAD

**Issue by gipert** [legend-exp/legend-data-format-specs#2]
Even if we don't expect to change it often, I propose to tag our LH5 format specification for data preservation / reproducibility. We should then store this version as LH5 file metadata.

See also: https://github.com/legend-exp/pygama/issues/315

**Comment by oschulz** [comment]
Yes, good point. Let's wait a little bit longer though until I've had time to update the writeup and docs build, Ok?

**Comment by jasondet** [comment]
What if we simply tag this repo and then specify that data objects should have an attribute, something like `rev`, set to the revision number in the tag? This would allow for schema evolution of the data_objects.

**Comment by oschulz** [comment]
I'm not sure tagging the repo will make a lot of sense, since it's supposed to cover all our data formats. As for LH5, we'll have to see if we need a version number or if "capabilities used" can be inferred from the file (should be the case currently). In any case we should document both older and newer formats in this repo if we have a format evolution, not just the latest format.

**Comment by jasondet** [comment]
I worry that requiring “capabilities used to be inferable from the file” would ultimately limit the possibilities for schema evolution.

I agree that documentation of all older formats must be part of newer specs, so that the new spec supports decoding of all available formats

**Comment by oschulz** [comment]
> I worry that requiring “capabilities used to be inferable from the file” would ultimately limit the possibilities for schema evolution.

We can definitely add an dataformat-version attribute to all datasets, in addition to `datatype`, defaulting to v1.0 if it's not present. That also means that we don't have to do it until we have a v1.1 or so (but we can, of course). :-)
